specifyweb/frontend/js_src/lib/components/QueryBuilder/ResultsWrapper.tsx

```diff
@@ -196,15 +198,20 @@ export function useQueryResultsWrapper({
               ? (fieldSpec, sortType): void => {
                   /*
                    * If some fields are not displayed, visual index and actual field
-                   * index differ
+                   * index differ. Also needs to skip phantom fields (added by locality)
                    */
                   const index = fieldSpecs.indexOf(fieldSpec);
-                  const field = displayedFields[index];
+                  const displayField = displayedFields[index];
+                  const lineIndex = allFields.indexOf(displayField);
                   handleSortChange(
-                    replaceItem(allFields, index, {
-                      ...field,
-                      sortType,
-                    })
+                    replaceItem(
+                      allFields.filter((field) => !queryFieldIsPhantom(field)),
```

Comment:
(optional) instead of adding phantom fields and then trying to remove them, you can try using the original fields array before more fields are added

I haven't looked to deep in the code to see if that is possible (if types match), but you can try to take a look around

Reply:
In the case of SpAuditLog query, unhidden fields will be added. On sorting in this case, we do want those fields so we cannot use the original fields directly (as they wouldn't contain additional sp auditlog query fields)